requirements.txt

```diff
@@ -0,0 +1,15 @@
+cycler==0.10.0
```

Comment:
The depenencies given by requirements.txt are just plain wrong! The code does not use most of the packages listed.

Reply:
I'd like to use a package handler such as anaconda which is used broadly in many companies to avoid module conflicts.

Reply:
@sebastianriese 
Actually, these packages are required for running the dashboard.
`cycler` dependency is pulled in from `matplotlib`
```
Requirement already satisfied: cycler>=0.10 in ./myvenv/lib/python3.4/site-packages (from matplotlib>=1.3.1->scikit-image)
```
I'm pretty sure this is the minimal set of required packages to get the dashboard up and running. But it's easy to check, just setup a new virtualenv and try running the dashboard.

@JeanRintoul :
As far as an anaconda based package handling, I think a `requirements.txt` file can help bootstrap a project. And it's incredibly useful (and basically required) for anyone _not_ using anaconda.

Reply:
I fully agree we should also support people who do not use anaconda (for example using anaconda would be an inconvenience for me, most of my python packages simply come from my Linux distribution, to use anaconda I would have to install things twice and switch between Python environments). But I think we should only add *immediate* dependencies to our requirements, that is those packages that we import directly. Adding transitive dependencies is a can of worms we do not want to open! We simply cannot keep up with the dependency changes of our dependencies; that's what package managers are for, pip should be quite able to handle that by itself, as are the package managers of any decent distribution, also I am strongly against requiring specific versions with `==`, this is overly specific and hurtful, we should use `~=` (compatible version) unless there is known breakage (or that version is antique). In short: requirements.txt should only contain the packages that we use directly and should not prescribe specific versions, doing so will deter people from installing and using the software.

Reply:
>we should only add immediate dependencies to our requirements
+1 If an immediate dependency drops one of *its* dependencies, there's no reason for users to install what is now essentially a random package.

>I am strongly against requiring specific versions with `==`
+1 If anyone wants to install this package with the same env as other packages, and another package with the same dependency, you can create unnecessary and difficult to resolve version conflicts by being overly specific on versioning.